sys/phydat/phydat_str.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_sensif
+ * @{
+ *
+ * @file
+ * @brief       Generic sensor/actuator data handling
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+
+#include "phydat.h"
+
+void phydat_dump(phydat_t *data, int dim)
+{
+    if (data == NULL || dim < 0 || dim > PHYDAT_DIM) {
+        puts("Unable to display data object");
+        return;
+    }
+    printf("Data:");
+    for (int i = 0; i < dim; i++) {
+        printf("\t[%i] - %i%s%s\n", i, data->val[i],
+               phydat_scale_to_str(data->scale),
+               phydat_unit_to_str(data->unit));
+    }
+}
+
+const char *phydat_unit_to_str(uint8_t unit)
+{
+    switch (unit) {
+        case UNIT_DEG_C:    return "°C";
+        case UNIT_DEG_F:    return "°F";
+        case UNIT_DEG_K:    return "°K";
```

Comment:
"Unlike the degree Fahrenheit and degree Celsius, the kelvin is not referred to or typeset as a degree"